packages/react/src/components/IconButton/index.tsx

```diff
@@ -13,55 +13,95 @@ import {
   PolymorphicProps,
   PolymorphicComponent
 } from '../../utils/polymorphicComponent';
-import { HTMLAttributes } from 'enzyme';
 
 export interface IconButtonProps
-  extends PolymorphicProps<React.HTMLAttributes<HTMLButtonElement>, 'button'> {
+  extends PolymorphicProps<
+    React.HTMLAttributes<HTMLButtonElement | HTMLAnchorElement>,
+    'button' | 'a'
```

Comment:
When writing tests, the type was a bit too restrictive in the sense that we do support both `button`, `a`, and linkish elements. We could address separately if needed but I don't think this is a breaking change as it's widening the type.